WordPress/gutenberg
# QuickEdit: Add Featured Image Control

## What?

Part of #64519 

This PR adds the Featured Image Control for the Page view. 



https://github.com/user-attachments/assets/cc8d5a58-97fd-4ac5-b1cc-ed705f669bd4



## View mode

| Grid | Table | Layout |
|--------|--------|--------|
| ![image](https://github.com/user-attachments/assets/2d8820a6-61e7-4bcf-ada5-89170c4f960a) | ![image](https://github.com/user-attachments/assets/7ab4799d-8439-4a8f-ab9c-6e1179413583) | ![image](https://github.com/user-attachments/assets/7634d2bb-9823-4133-be9f-8b009ad3e6f5) | 


## Edit

| Without Image | With Image |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/ae8c22cd-b55c-49e9-b953-3fea9e153ec1) | ![image](https://github.com/user-attachments/assets/222358eb-14b8-42ea-b2ef-dd332d402d02) | 


## Additional Changes

### Render component and view prop

With this PR, the render component will now receive the view prop. This addition is necessary because the component’s layout needs to adapt based on the view. Similar cases include the Title field, where different layouts are required depending on the view.

https://github.com/WordPress/gutenberg/blob/83569778dba9dd64dbd6db2e4623e36457b7f7ac/packages/edit-site/src/components/post-fields/index.js#L163-L164

### Fix error on bulk edit mode

https://github.com/user-attachments/assets/c906cb68-9328-4cca-b61d-f24abf1f8861

```
Uncaught TypeError: Cannot read properties of null (reading 'status')
```

Currently, there is an error when the user edits a field in bulk mode. [This PR fixes this error](https://github.com/WordPress/gutenberg/pull/64496/files#diff-4ed486e973856f952d2b80aea381fc1f5a1d0004995a0b68554147b02cee0b55R75).

### Attachment type

This PR updates the `Attachment` type.







## Why?


## How?


## Testing Instructions





Ensure that `Quick Edit in DataViews` and `Quick Edit in DataViews` are enabled.

1. Visit the Site Editor
2. Click on pages.
3. Toggle to the table view.
4. Click on the settings.
5. Make the feature image field visible.
6. Click on details panel icon.
7. Ensure that the Featured Image option is available.
8. Click on it.
9. Ensure that the Media Gallery opens and click on an image.
10. Save.
11. Ensure that the image is set as the featured image to the post.
12. Toggle to the grid view.
13. Ensure that the image is rendered correctly.
14. Toggle to the list view.
15. Ensure that the image is rendered correctly.



### Testing Instructions for Keyboard


## Screenshots or screencast
